core/arch/arm/include/arm32_macros.S

```diff
@@ -123,10 +188,30 @@
 	mrc	p15, 0, \reg, c13, c0, 1
 	.endm
 
+	.macro write_tpidruro reg
+	mcr	p15, 0, \reg, c13, c0, 3
+	.endm
+
+	.macro read_tpidruro reg
+	mrc	p15, 0, \reg, c13, c0, 3
+	.endm
+
+	.macro write_diag reg
+	mcr  p15, 0, \reg, c15, c0, 1
```

Comment:
All the `c15` registers are implementation defined, that is, defined based on CPU type.
Perhaps we can keep them in a `arm32_macros_cortex_a??.S` etc as needed.